web/api/formatters/rrd2json.c

```diff
@@ -22,6 +22,58 @@ static inline void free_temp_rrddim(RRDDIM *temp_rd)
     }
 }
 
+void free_context_param_list(struct context_param **param_list)
+{
+    if (unlikely(!param_list || !*param_list))
+        return;
+
+    free_temp_rrddim(((*param_list)->rd));
+    freez((*param_list));
+    *param_list = NULL;
+}
+
+void build_context_param_list (struct context_param **param_list, RRDSET *st)
+{
+    if (unlikely(!param_list || !st))
+        return;
+
+    if (unlikely(!(*param_list))) {
+        *param_list = mallocz(sizeof(struct context_param));
+        (*param_list)->first_entry_t = LONG_MAX;
+        (*param_list)->last_entry_t = 0;
+        (*param_list)->last_accessed_time = now_realtime_sec();
+        (*param_list)->rd = NULL;
+    }
+
+    RRDDIM *rd1;
+    rrdset_rdlock(st);
+
+    st->last_accessed_time = (*param_list)->last_accessed_time;
```

Comment:
The loop that calls `build_context_param_list()` can take many seconds depending on the load of the computer. As a result, the `last_accessed_time` can be a minute old, and in the meantime this chart may have been queried many times since then.